sw/device/lib/drivers/rv_plic/rv_plic.c

```diff
@@ -34,337 +81,417 @@ typedef struct plic_reg_info {
-                                     dif_plic_irq_id_t *claim_data) {
-  if (plic == NULL || target >= RV_PLIC_PARAM_NUM_TARGET ||
-      claim_data == NULL) {
+
+plic_result_t plic_irq_claim(plic_target_t target, plic_irq_id_t *claim_data) {
+  if (target >= RV_PLIC_PARAM_NUM_TARGET || claim_data == NULL) 
+  {
     return kDifPlicBadArg;
   }
-
-  ptrdiff_t claim_complete_reg = plic_claim_complete_base_for_target(target);
-  *claim_data = mmio_region_read32(plic->params.base_addr, claim_complete_reg);
+  
+  *claim_data = rv_plic_peri->CC0;
 
   return kDifPlicOk;
 }
 
-dif_plic_result_t dif_plic_irq_complete(
-    const dif_plic_t *plic, dif_plic_target_t target,
-    const dif_plic_irq_id_t *complete_data) {
-  if (plic == NULL || target >= RV_PLIC_PARAM_NUM_TARGET ||
-      complete_data == NULL) {
+
+plic_result_t plic_irq_complete(const plic_irq_id_t *complete_data) {
+  if (complete_data == NULL) 
+  {
     return kDifPlicBadArg;
   }
 
   // Write back the claimed IRQ ID to the target specific CC register,
   // to notify the PLIC of the IRQ completion.
-  ptrdiff_t claim_complete_reg = plic_claim_complete_base_for_target(target);
-  mmio_region_write32(plic->params.base_addr, claim_complete_reg,
-                      *complete_data);
+  rv_plic_peri->CC0 = *complete_data;
 
   return kDifPlicOk;
 }
 
-dif_plic_result_t dif_plic_software_irq_force(const dif_plic_t *plic,
-                                              dif_plic_target_t target) {
-  if (plic == NULL || target >= RV_PLIC_PARAM_NUM_TARGET) {
-    return kDifPlicBadArg;
-  }
 
-  ptrdiff_t msip_offset = plic_software_irq_base_for_target(target);
-  mmio_region_write32(plic->params.base_addr, msip_offset, 1);
+plic_result_t plic_software_irq_force()
+{
+  rv_plic_peri->MSIP0 = 1;
 
   return kDifPlicOk;
 }
 
-dif_plic_result_t dif_plic_software_irq_acknowledge(const dif_plic_t *plic,
-                                                    dif_plic_target_t target) {
-  if (plic == NULL || target >= RV_PLIC_PARAM_NUM_TARGET) {
-    return kDifPlicBadArg;
-  }
 
-  ptrdiff_t msip_offset = plic_software_irq_base_for_target(target);
-  mmio_region_write32(plic->params.base_addr, msip_offset, 0);
+plic_result_t plic_software_irq_acknowledge()
+{
+  rv_plic_peri->MSIP0 = 0;
 
   return kDifPlicOk;
 }
 
-dif_plic_result_t dif_plic_software_irq_is_pending(const dif_plic_t *plic,
-                                                   dif_plic_target_t target,
-                                                   bool *is_pending) {
-  if (plic == NULL || target >= RV_PLIC_PARAM_NUM_TARGET ||
-      is_pending == NULL) {
-    return kDifPlicBadArg;
-  }
-
-  ptrdiff_t msip_offset = plic_software_irq_base_for_target(target);
-  uint32_t register_value =
-      mmio_region_read32(plic->params.base_addr, msip_offset);
 
-  *is_pending = (register_value == 1) ? true : false;
+plic_result_t plic_software_irq_is_pending(bool *is_pending)
```

Comment:
do not waste one reference pass + return variable, just use the return variable to say true or false